### PR TITLE
[WIP] Add `is_hermitian` property to `Operator` 

### PR DESCRIPTION
**Pull request by Jaybsoni**
**Context:**
Typically we use the type of an object to determine whether it can be used in a `MeasurementProcess`. We explicitly check if the type is `Observable`. This conflicts with operator-arithmetic classes which may or may not be observables depending on the instance. (eg. a `Sum` of observables may not be an observable unless their coefficients were real)  

**Description of the Change:**
Add a property `is_hermitian` to the base `Operator` class, use this to determine if an operator can be used in a `MeasurementProcess`.

### CODE REVIEW

File: pennylane/measurements.py [hunk 1]
```diff
@@ -463,7 +463,8 @@ def circuit(x):
     Raises:
         QuantumFunctionError: `op` is not an instance of :class:`~.Observable`
     """
-    if not isinstance(op, (qml.operation.Observable, qml.Hamiltonian)):
+    if not op.is_hermitian:
+    # if not isinstance(op, (qml.operation.Observable, qml.Hamiltonian)):
```
Comment (Jaybsoni) [inline pennylane/measurements.py:467]:
```suggestion
```

File: pennylane/measurements.py [hunk 2]
```diff
@@ -498,7 +499,8 @@ def circuit(x):
     Raises:
         QuantumFunctionError: `op` is not an instance of :class:`~.Observable`
     """
-    if not isinstance(op, qml.operation.Observable):
+    if not op.is_hermitian:
+    # if not isinstance(op, qml.operation.Observable):
```
Comment (Jaybsoni) [inline pennylane/measurements.py:503]:
```suggestion
```

File: pennylane/measurements.py [hunk 3]
```diff
@@ -577,7 +579,8 @@ def circuit(x):
         observable ``obs``.
     """
     if (
-        not isinstance(op, qml.operation.Observable) and op is not None
+        op is not None and not op.is_hermitian
+        # not isinstance(op, qml.operation.Observable) and op is not None
```
Comment (Jaybsoni) [inline pennylane/measurements.py:583]:
```suggestion
```

**Review by albi3ro — APPROVED** [review]
Looks good 👍

**Pull request merged**